client/my-sites/stats/activity-log-search/index.jsx

```diff
@@ -0,0 +1,112 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { omit, isEmpty } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ActivityLogSearchTokens from './activity-log-search-tokens';
+import ActivityLogSearchGroups from './activity-log-search-groups';
+import Button from 'components/button';
+import Gridicon from 'gridicons';
+
+class ActivityLogSearch extends Component {
+	static propTypes = {
+		translate: PropTypes.func.isRequired,
+		filter: PropTypes.object.isRequired,
+	};
+
+	state = {
+		activeFilter: null,
+	};
+
+	hasTokens() {
+		return ! isEmpty( omit( this.props.filter, 'page' ) );
```

Comment:
this won't matter that much but it's still a relatively costly operation for what we're wanting to know. something that would end up maybe much cheaper would be a variant that doesn't clone the filter and omit the props. also, I think we want to make sure that we use the arrow-function here to prevent `this` binding from going astray

```js
hasTokens = () => {
	const { filter } = this.props;
	const keyCount = Object.keys( filter );

	return keyCount > 1 || ( ! filter.hasOwnProperty( 'page' ) && keyCount > 0 );
}
```

this is ugly and unclear though. if the `page` is causing us enough grief we can pull it out of the `filter`, especially if it's eventually stripped from the query URL.

Reply:
because of the complex logic here it could make sense to encapsulate some of the logic inside the state itself on update. that is, every time we update the filter we add certain properties like if it's empty…

```js
this.props.filter.isEmpty
```